src/graphics/PlanningView.ts

```diff
@@ -483,88 +335,168 @@ export function drawHealthBarAboveHead(unitIndex: number, underworld: Underworld
-      const healthBarHealColor = u.faction == Faction.ALLY ? 0x23ff30 : 0xff2828;
-      globalThis.unitOverlayGraphics?.lineStyle(0, 0x000000, 1.0);
-      globalThis.unitOverlayGraphics?.beginFill(healthBarColor, 1.0);
-      const healthBarProps = getUIBarProps(u.x, u.y, u.health, u.healthMax, zoom, u);
-      globalThis.unitOverlayGraphics?.drawRect(
-        healthBarProps.x,
-        // Stack the health bar above the mana bar
-        healthBarProps.y - config.UNIT_UI_BAR_HEIGHT / zoom,
-        healthBarProps.width,
-        healthBarProps.height
-      );
+    if (u.x !== null && u.y !== null && u.alive && !globalThis.isHUDHidden) {
 
-      // Only show health bar predictions on PlayerTurns, while players are able
-      // to cast, otherwise it will show out of sync when NPCs do damage
-      if (underworld.turn_phase == turn_phase.PlayerTurns && globalThis.unitOverlayGraphics) {
-        // Show how much damage they'll take on their health bar
-        globalThis.unitOverlayGraphics.beginFill(healthBarHurtColor, 1.0);
-        if (predictionUnit) {
-          const healthAfterHurt = predictionUnit.health;
-          if (healthAfterHurt > u.health) {
-            globalThis.unitOverlayGraphics.beginFill(healthBarHealColor, 1.0);
-          }
-          // const healthBarHurtWidth = Math.max(0, config.UNIT_UI_BAR_WIDTH * (u.health - healthAfterHurt) / u.healthMax);
-          const healthBarHurtProps = getUIBarProps(u.x, u.y, u.health - healthAfterHurt, u.healthMax, zoom, u);
-          globalThis.unitOverlayGraphics.drawRect(
-            // Show the healthBarHurtBar on the right side of the health  bar
-            healthBarHurtProps.x + config.UNIT_UI_BAR_WIDTH / zoom * healthAfterHurt / u.healthMax,
-            // Stack the health bar above the mana bar
-            healthBarHurtProps.y - config.UNIT_UI_BAR_HEIGHT / zoom,
-            healthBarHurtProps.width,
-            healthBarHurtProps.height);
-          // Draw red death circle if a unit is currently alive, but wont be after cast
-          if (u.alive && !predictionUnit.alive) {
-            const skullPosition = withinCameraBounds({ x: u.x, y: u.y - config.COLLISION_MESH_RADIUS * 2 + 8 });
-            const imagePath = globalThis.player && u.faction === globalThis.player.unit.faction ? 'badgeDeathAlly.png' : 'badgeDeath.png';
-            ImmediateMode.draw(imagePath, skullPosition, (1 / zoom) + (Math.sin(Date.now() / 500) + 1) / 3);
+      // Draw base health bar
+      if (globalThis.unitOverlayGraphics) {
+        const healthBarColor = u.faction == Faction.ALLY ? colors.healthAllyGreen : colors.healthRed;
+        const healthBarHurtColor = u.faction == Faction.ALLY ? colors.healthAllyDarkGreen : colors.healthDarkRed;
+        const healthBarHealColor = u.faction == Faction.ALLY ? colors.healthAllyBrightGreen : colors.healthBrightRed;
+
+        const healthBarMax = u.healthMax || 1;
+        //background
```

Comment:
The healthbar background seems to apply to the player but not to golems

Reply:
Edit: I was wrong, it does show on NPCs but for the player the background color is the samee as the "Prediction Loss Health" color whereas for the NPCs it's not.  I think it'd be best if there were a different color for each.  For allies, real health is the bright green, prediction loss health is the dark green and then missing health could be the very dark grey (non colored). And for enemies real health is red, prediction loss health is dark red and missing health (background) is dark grey?

Reply:
The background is dark grey currentlt. I think what you're seeing is the transparency, making it mix with color behind it.

I thought it looked a little strange too. Now I'm more confident it making it somewhat (or completely) opaque.

Also, you might've noticed non-mana units do not have a mana bar background. And if a non-mana unit does get mana, it will show as a full bar. I thought that was especially nice for send mana 🙂